### PR TITLE
Charlesmchen/profiles9a

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/Profiles/OWSProfileManager.m [hunk 1]
```diff
@@ -423,18 +423,20 @@ - (void)updateProfileOnService:(nullable NSString *)localProfileName
...
+                                                                           avatarDigest:avatarDigest];
+
+        [self.networkManager makeRequest:request
+            success:^(NSURLSessionDataTask *task, id responseObject) {
+                successBlock();
+            }
+            failure:^(NSURLSessionDataTask *task, NSError *error) {
+                DDLogError(@"%@ Failed to update profile with error: %@", self.tag, error);
+                failureBlock();
+            }];
```
Comment (charlesmchen) [inline Signal/src/Profiles/OWSProfileManager.m]: Actually update profiles (only works against staging at the moment).

File: Signal/src/Profiles/ProfileFetcherJob.swift [hunk 2]
```diff
@@ -177,26 +175,26 @@ struct SignalServiceProfile {
...
             }
-            avatarDigestEncrypted = data
+            avatarDigest = data
         }
 
         // `removeKeyType` is an objc category method only on NSData, so temporarily cast.
         self.identityKey = (identityKeyWithType as NSData).removeKeyType() as Data
         self.profileNameEncrypted = profileNameEncrypted
-        self.avatarUrlEncrypted = avatarUrlEncrypted
-        self.avatarDigestEncrypted = avatarDigestEncrypted
```
Comment (charlesmchen) [inline Signal/src/Profiles/ProfileFetcherJob.swift]: avatar fields are not encrypted.

File: Signal/src/Profiles/ProfileFetcherJob.swift [hunk 3]
```diff
@@ -177,26 +175,26 @@ struct SignalServiceProfile {
             profileNameEncrypted = data
         }
 
-        var avatarUrlEncrypted: Data? = nil
+        var avatarUrlData: Data? = nil
         if let avatarUrlString = responseDict["avatar"] as? String {
```
Comment (michaelkirk) [inline Signal/src/Profiles/ProfileFetcherJob.swift]: Is this really a base64 encoded string? I assumed the server was sending UTF-8 and the decoding we're doing in profile manager is unnecessary.

File: SignalServiceKit/src/Messages/Interactions/TSOutgoingMessage.m [hunk 4]
```diff
@@ -455,10 +455,9 @@ - (OWSSignalServiceProtosDataMessageBuilder *)dataMessageBuilder
     return builder;
 }
 
-- (OWSSignalServiceProtosDataMessage *)buildDataMessage:(NSString *)recipientId
+- (OWSSignalServiceProtosDataMessage *)buildDataMessage:(NSString *_Nullable)recipientId
```
Comment (michaelkirk) [inline SignalServiceKit/src/Messages/Interactions/TSOutgoingMessage.m]: When is this null? Maybe deserves a comment in the function header.

File: SignalServiceKit/src/TSConstants.h [hunk 5]
```diff
@@ -41,6 +41,7 @@ typedef enum { kSMSVerification, kPhoneNumberVerification } VerificationTranspor
 #define textSecureDeviceProvisioningAPIFormat @"v1/provisioning/%@"
 #define textSecureDevicesAPIFormat @"v1/devices/%@"
 #define textSecureProfileAPIFormat @"v1/profile/%@"
+#define textSecureSetProfileAPI @"v1/profile"
```
Comment (michaelkirk) [inline SignalServiceKit/src/TSConstants.h]: Feel free to reimplement all the request building ceremony with a request factory any day you're bored. =)

File: Signal/src/Profiles/ProfileFetcherJob.swift (hunk 3 above)
Comment (charlesmchen) [inline Signal/src/Profiles/ProfileFetcherJob.swift]: Ask Moxie.

File: SignalServiceKit/src/Messages/Interactions/TSOutgoingMessage.m (hunk 4 above)
Comment (charlesmchen) [inline SignalServiceKit/src/Messages/Interactions/TSOutgoingMessage.m]: Done.

File: SignalServiceKit/src/TSConstants.h (hunk 5 above)
Comment (charlesmchen) [inline SignalServiceKit/src/TSConstants.h]: =)

**Pull request merged**